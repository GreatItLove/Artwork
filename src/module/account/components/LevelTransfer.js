import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Field } from 'redux-form';
import { RaisedButton, MenuItem } from 'material-ui';
import _map from 'lodash/map';

import SelectField from '../../common/components/form/SelectField';

const propTypes = {
  t: PropTypes.func.isRequired,
  list: PropTypes.array.isRequired,
  account: PropTypes.object.isRequired,
  resetTransferEmail: PropTypes.func.isRequired
};

/**
 * @class LevelTransfer
 * @extends React.Component
 * @description Render component
 */
class LevelTransfer extends Component {
  state = {
    isOpen: false
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.open !== this.props.open) {
      this.setState({ isOpen: false })
    }
  }
  renderTransferFields() {
    const { isOpen } = this.state;
    const { list, account } = this.props;
    const levelClass = classNames({
      'users levelTransfer': true,
      'filter-open users levelBorder': isOpen,
      'filter-hidden': !isOpen,
    });
    return (
      <div
        className={levelClass}
        style={{ height: isOpen ? 250 : 0 }}>
        <div className="col-sm-12">
          <div className="row align-items-center">
            <div className="col-md-4" />
            <div className="col-md-8">
              <div className="py-3">
                Every Artwok Manager has one Site Administrator user with the
                highest access level. Right now, you are the S.A. This may be
                transferred away to another level 4 user, but{' '}
                <b>this cannot be undone.</b>
              </div>
              <div>To give full control of your Artwork Manager to a different user choose below:</div>
              <Field id="transferEmail" name="transferEmail" fullWidth hintText="Select One" component={SelectField}>
                {_map(
                  list,
                  item =>
                    item.userLevel === 4 && account.email !== item.email ? (
                      <MenuItem key={item.userId} value={item.userId} primaryText={item.email} />
                    ) : null
                )}
              </Field>
            </div>
          </div>
        </div>
      </div>
    );
  }
  openTransferBlock = () => {
    this.setState({ isOpen: !this.state.isOpen });
    this.props.resetTransferEmail('transferEmail');
  };
  /**
   * @description render
   * @return {Object} JSX HTML Content
   */
  render() {
    const { t, account } = this.props;
    const { isOpen } = this.state;
    return account.userType === 'SA' && account.userLevel === 4 ? (
      <div className="pt-2">
        <div className="col-sm-12 py-3">
          <div className="row align-items-center">
            <div className="col-md-4">
              <label>{t('editAccount.transfer')}</label>
            </div>
            <div className="col-md-8">
              <RaisedButton
                backgroundColor={isOpen ? '#8f9082' : '#dddddd'}
                labelStyle={{ color: isOpen ? '#ffffff' : '#000000', textTransform: 'none' }}
                onClick={this.openTransferBlock}
                label={t('editAccount.siteAdmin')}
              />
            </div>
          </div>
        </div>
        {this.renderTransferFields()}
      </div>
    ) : null;
  }
}

LevelTransfer.propTypes = propTypes;

export default LevelTransfer;
