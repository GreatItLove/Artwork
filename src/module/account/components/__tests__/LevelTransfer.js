import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import LevelTransfer from '../LevelTransfer';

const account = {
  active: true,
  email: 'ahershey79@yahoo.com',
  fullName: 'Andrew Hershey',
  isFetchedData: true,
  loadingData: false,
  loadingError: false,
  userId: 6757,
  userLevel: 4,
  userType: 'SA'
};
const list = [
  {
    active: true,
    email: 'jawahar.n@mitrahsoft.com',
    fullName: 'jawahar',
    lastLogin: '',
    userId: 6842,
    userLevel: 4,
    userType: 'Admin'
  },
  {
    active: true,
    email: 'testlasd@dsgagjads.com',
    fullName: 'Tets Test22',
    lastLogin: '',
    userId: 6846,
    userLevel: 1,
    userType: 'Guest'
  }
];
const props = {
  list: list,
  account: account,
  resetTransferEmail: jest.fn(),
  t: key => key
};

describe('LevelTransfer', () => {
  it('Should be same snapshot', () => {
    const wrapper = shallow(<LevelTransfer {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
