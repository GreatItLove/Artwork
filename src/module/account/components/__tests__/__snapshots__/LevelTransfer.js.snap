// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelTransfer Should be same snapshot 1`] = `
<div
  className="pt-2"
>
  <div
    className="col-sm-12 py-3"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-md-4"
      >
        <label>
          editAccount.transfer
        </label>
      </div>
      <div
        className="col-md-8"
      >
        <RaisedButton
          backgroundColor="#dddddd"
          disabled={false}
          fullWidth={false}
          label="editAccount.siteAdmin"
          labelPosition="after"
          labelStyle={
            Object {
              "color": "#000000",
            }
          }
          onClick={[Function]}
          primary={false}
          secondary={false}
        />
      </div>
    </div>
  </div>
  <CSSTransition
    appear={true}
    classNames="fade"
    in={false}
    mountOnEnter={true}
    timeout={300}
    unmountOnExit={true}
  >
    <div
      className="col-sm-12"
      style={
        Object {
          "borderBottom": "1px solid #82b5cb",
          "borderTop": "1px solid #82b5cb",
        }
      }
    >
      <div
        className="row align-items-center"
      >
        <div
          className="col-md-4"
        />
        <div
          className="col-md-8"
        >
          <div
            className="py-3"
          >
            Every Artwok Manager has one Site Administrator user with the highest access level. Right now, you are the S.A. This may be transferred away to another level 4 user, but 
            <b>
              this cannot be undone.
            </b>
          </div>
          <div>
            To give full control of your Artwork Manager to a different user choose below:
          </div>
          <Field
            component={[Function]}
            fullWidth={true}
            hintText="Select One"
            id="transferEmail"
            name="transferEmail"
          >
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="jawahar.n@mitrahsoft.com"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value={6842}
            />
          </Field>
        </div>
      </div>
    </div>
  </CSSTransition>
</div>
`;
